src/components/vm/consoles/common.jsx

```diff
@@ -0,0 +1,170 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import cockpit from 'cockpit';
+
+import { Button } from "@patternfly/react-core/dist/esm/components/Button";
+import { Content, ContentVariants } from "@patternfly/react-core/dist/esm/components/Content";
+import { ClipboardCopy } from "@patternfly/react-core/dist/esm/components/ClipboardCopy/index.js";
+import { Flex } from "@patternfly/react-core/dist/esm/layouts/Flex";
+import { DownloadIcon } from "@patternfly/react-icons";
+import { fmt_to_fragments } from 'utils.jsx';
+import { InfoPopover } from '../../common/infoPopover';
+
+import { domainDesktopConsole } from '../../../libvirtApi/domain.js';
+import { StateObject } from './state';
+
+const _ = cockpit.gettext;
+
+export class ConsoleState extends StateObject {
+    constructor () {
+        super();
+        this.connected = true;
+    }
+
+    setConnected(val) {
+        this.connected = val;
+        this.update();
+    }
+}
+
+export function connection_address() {
+    let address;
+    if (cockpit.transport.host == "localhost") {
+        const app = cockpit.transport.application();
+        if (app.startsWith("cockpit+=")) {
+            address = app.substr(9);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2008-57844738-20250716-080045-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)